container/container.go

```diff
@@ -47,151 +64,409 @@ var statuses = []struct {
+				"mesg.engine":                c.engineName,
+				"mesg.service":               srv.Hash.String(),
+				"mesg.instance":              instanceHash.String(),
+				"mesg.runner":                runnerHash.String(),
+				"mesg.dependency":            service.MainServiceKey,
+				"com.docker.stack.namespace": runnerName,
+			},
+		},
+		TaskTemplate: swarm.TaskSpec{
+			ContainerSpec: &swarm.ContainerSpec{
+				Image: imageTag + srv.Hash.String(),
+				Labels: map[string]string{
+					"com.docker.stack.namespace": runnerName,
+				},
+				Args:    srv.Configuration.Args,
+				Command: strings.Fields(srv.Configuration.Command),
+				Env: xos.EnvMergeSlices(instanceEnv, []string{
+					"MESG_SERVICE_HASH=" + srv.Hash.String(),
+					"MESG_INSTANCE_HASH=" + instanceHash.String(),
+					"MESG_RUNNER_HASH=" + runnerHash.String(),
+					"MESG_ENDPOINT=" + c.engineEndpoint,
+					"MESG_REGISTER_SIGNATURE=" + base64.StdEncoding.EncodeToString(registerPayload),
+					"MESG_ENV_HASH=" + instanceEnvHash.String(),
+				}),
+				Mounts:          append(volumes, volumesFrom...),
+				StopGracePeriod: &stopGracePeriod,
+			},
+			Networks: []swarm.NetworkAttachmentConfig{
+				{
+					Target:  networkID,
+					Aliases: []string{service.MainServiceKey},
+				},
+				{
+					Target: enginedNetworkID,
+				},
+			},
+			RestartPolicy: &swarm.RestartPolicy{
+				Condition:   swarm.RestartPolicyConditionOnFailure,
+				MaxAttempts: &maxAttempts,
+			},
+		},
+		EndpointSpec: &swarm.EndpointSpec{
+			Ports: convertPorts(srv.Configuration.Ports),
+		},
+	})
+
+	// Start
+	for _, spec := range specs {
+		if err := c.startService(spec); err != nil {
+			c.Stop(srv, runnerHash)
+			return err
+		}
+	}
+
+	return nil
 }
 
-// CreateNetwork creates a Docker Network with a namespace. Retruns network id and error.
-func (c *DockerContainer) CreateNetwork(namespace string) (string, error) {
-	namespace = c.namespace(namespace)
-	network, err := c.client.NetworkInspect(context.Background(), namespace, types.NetworkInspectOptions{})
+// Stop stops an instance.
+func (c *Container) Stop(srv *service.Service, runnerHash hash.Hash) error {
+	var (
+		errs  xerrors.SyncErrors
+		name  = servicePrefix + runnerHash.String()
+		names = make([]string, 0)
+	)
+
+	for _, dep := range srv.Dependencies {
+		names = append(names, name+"_"+dep.Key)
+	}
+	names = append(names, name)
+
+	for _, name := range names {
+		if err := c.stopService(name); err != nil {
+			errs.Append(err)
+		}
+	}
+	if err := errs.ErrorOrNil(); err != nil {
+		return err
+	}
+
+	return c.deleteNetwork(name)
+}
+
+// deleteData deletes the data volumes of instance and its dependencies.
+// TODO: right now deleteData() is not compatible to work with multiple instances of same
+// service since extractVolumes() accepts service, not instance. same applies in the start
+// api as well. make it work with multiple instances.
+// func deleteData(cont container.Container, s *service.Service) error {
```

Comment:
This function is not used anymore so I had to comment it because of the lint.
I would like to keep it if we need it in the future.